src/packages/utils/src/things/json-rpc/json-rpc-data.ts

```diff
@@ -1,57 +1,107 @@
 import { BaseJsonRpcType } from "./json-rpc-base-types";
-import { strCache, toStrings } from "./json-rpc-base-types";
+const BUFFER_EMPTY = Buffer.allocUnsafe(0);
 
-function validateByteLength(byteLength?: number) {
-  if (typeof byteLength !== "number" || !(byteLength >= 0)) {
-    throw new Error(`byteLength must be a number greater than or equal to 0, provided: ${byteLength}`);
-  }
-}
+type JsonRpcDataInputArg = string | Buffer;
 
 export class Data extends BaseJsonRpcType {
-
-  constructor(value: string | Buffer, private _byteLength?: number) {
+  constructor(value: JsonRpcDataInputArg, private _byteLength?: number) {
     super(value);
+
     if (typeof value === "bigint") {
       throw new Error(`Cannot create a ${typeof value} as a Data`);
     }
     if (_byteLength !== undefined) {
```

Comment:
I think this check is redundant because we check it in the `validateByteLength` function

Reply:
Yes!